src/MgvOfferTaking.sol

```diff
@@ -202,20 +220,15 @@ abstract contract MgvOfferTaking is MgvHasOffers {
   {
     unchecked {
       if (
-        mor.fillVolume == 0 || sor.offerId == 0 || mor.maxRecursionDepth == 0 || sor.offer.logPrice() > mor.maxLogPrice
+        mor.fillVolume == 0 || sor.offer.logPrice() > mor.maxLogPrice || sor.offerId == 0 || mor.maxRecursionDepth == 0
+          || mor.gasreqForFailingOffers > mor.maxGasreqForFailingOffers
       ) {
```

Comment:
`MgvReader` should be updated to reflect this.

Reply:
How much of it? "MAxGasreqForFailingOffers" for instance does not make sense since MgvReader assumes success, right?

Reply:
Indeed it assumes success. 

The MgvReader implementation closely mirrors the MgvOfferTaking implementation for ease of validation and maintenance; This PR changes the structure of the code and I think MgvReader should be updated to reflect that change.

Reply:
Done